lib/vanagon/engine/amazon.rb

```diff
@@ -0,0 +1,69 @@
+require 'aws-sdk'
+require 'erb'
+require 'base64'
+require 'vanagon/engine/base'
+
+class Vanagon
+  class Engine
+    class Amazon < Base
+      attr_accessor :ami, :key_name, :userdata, :key, :key_name, :shutdown_behavior
+      attr_accessor :subnet_id, :instance_type, :vpc_id
+
+      def initialize(platform, target = nil)
+        super
+        @name = 'amazon'
+        @platform = platform
+        @ami = @platform.aws_ami
+        @target_user = @platform.target_user
+        @subnet_id = @platform.aws_subnet_id
+        @userdata = @platform.aws_user_data
+        @region = @platform.aws_region.nil? ? 'us-east-1' : @platform.region
+        @key_name = @platform.aws_key_name
+        @key = @platform.aws_key
+        @vpc_id = @platform.aws_vpc_id
+        @instance_type = @platform.aws_instance_type.nil? ? "t1.micro" : @platform.aws_instance_type
+        @required_attributes = ["ssh_port", "aws_ami", "aws_key_name"]
+        @shutdown_behavior = @platform.aws_shutdown_behavior
+
+        @ec2 = ::Aws::EC2::Client.new region: @region
+        @resource = ::Aws::EC2::Resource.new client: @ec2
+      end
+
+      def get_userdata()
+        unless @userdata.nil?
+          Base64.encode64(ERB.new(@userdata).result(binding))
+        end
+      end
+
+      def select_target
+        @instances = @resource.create_instances({
+          image_id: @ami,
+          min_count: 1,
+          max_count: 1,
+          key_name: @key_name,
+          instance_type: @instance_type,
+          subnet_id: @subnet_id,
+          user_data: get_userdata || "",
+          monitoring: {
+            enabled: false,
+          }
+        })
+
+        begin
+          puts "Created instance waiting for it to start up"
+          @ec2.wait_until(:instance_status_ok, instance_ids: @instances.map(&:id))
+          puts "Instance created id: #{@instances.map(&:id).first}"
+        rescue Aws::Waiters::Errors::WaiterFailed => error
```

Comment:
The calls to `Aws` so far have been fully qualified (`::Aws`); should this one be fully qualified too?